v2/internal/genericarmclient/arm_ids.go

```diff
@@ -73,3 +73,13 @@ func MakeResourceGroupID(subscription string, resourceGroup string) string {
 func MakeSubscriptionID(subscription string) string {
 	return fmt.Sprintf("/subscriptions/%s", subscription)
 }
+
+func GetSubscription(path string) (string, error) {
```

Comment:
ah! This is just moved from [here ](https://github.com/Azure/azure-service-operator/pull/2559/files/92bb540d5443e2a3039e953e1f634a7884a34294#diff-0614b9ea6824ce28ae2dc1bc07900f2929014ee58a6b4cb2da29f5a52f6d3521L100). Just wanted for it to be at the right place and be exported to be re-used.

Reply:
Yeah, I mean honestly you could probably not even bother with the `GetSubscription` function and instead just call `arm.ParseResourceID` right there, so that you have access to other stuff like `ResourceGroup` too if you need it (which as we discussed, you might actually).